File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(0, len(nums)):
            if target - nums[i] in nums:
                if nums[i] == target - nums[i] and nums.count(nums[i]) >= 2:
                    return [i, nums.index(nums[i],i+1)]
                elif nums[i] == target - nums[i] and nums.count(nums[i]) < 2: continue
                else:return [i, nums.index(target-nums[i])]
                
        

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map< int, int>omap;
        int i =0;
        for(i; i<nums.size(); i++){
            if(omap.count(target-nums[i])){
                return vector<int> {omap[target-nums[i]], i};
            }
            else{
                omap[nums[i]] = i;
            }
        }
        return vector<int> {omap[target-nums[i]], i};
    }
};